site/src/Controller/NotificationController.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Mgilet\NotificationBundle\Manager\NotificationManager;
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+/**
+ * Class NotificationController
+ */
+class NotificationController extends AbstractController
+{
+    /**
+     * @var NotificationManager
+     */
+    protected $notificationManager;
+
+    /**
+     * NotificationController constructor.
+     * @param NotificationManager $notificationManager
+     */
+    public function __construct(NotificationManager $notificationManager)
+    {
+        $this->notificationManager = $notificationManager;
+    }
+
+    /**
+     * Mark a Notification as seen
+     *
+     * @Route("/notification/seen", name="notification_mark_seen")
+     *
+     * @Method("POST")
+     *
+     * @param Request $request
+     *
+     * @throws \Doctrine\ORM\OptimisticLockException
+     * @throws \Doctrine\ORM\NonUniqueResultException
+     * @throws \Doctrine\ORM\EntityNotFoundException
+     *
+     * @return JsonResponse
+     */
+    public function markAsSeenAction(Request $request)
+    {
+        if ($request->isXmlHttpRequest()) {
+            $notification = $request->get('notification');
+            $notifiable = $request->get('notifiable');
+            $this->notificationManager->markAsSeen(
+                $this->notificationManager->getNotifiableInterface($this->notificationManager->getNotifiableEntityById($notifiable)),
+                $this->notificationManager->getNotification($notification),
+                true
+            );
+
+            return new JsonResponse('seen');
+        }
+
+        return new JsonResponse('Not an ajax request');
+    }
+
+    /**
+     * Remove a Notification
+     *
+     * @Route("/notification/remove", name="remove_notification")
+     *
+     * @Method("POST")
+     *
+     * @param Request $request
+     *
+     * @throws \Doctrine\ORM\OptimisticLockException
+     *
+     * @return JsonResponse
+     */
+    public function removeAction(Request $request)
+    {
+        if ($request->isXmlHttpRequest()) {
+            $notification = $request->get('notification');
+            $notifiable = $request->get('notifiable');
+
+            // Deletes the link between a Notifiable and a Notification
+            $this->notificationManager->removeNotification(
+                [$this->notificationManager->getNotifiableInterface($this->notificationManager->getNotifiableEntityById($notifiable))],
+                $this->notificationManager->getNotification($notification),
+            );
```

Comment:
Is this needed, isn't the deleteNotification() call below sufficient? If so, then the database schema is missing a couple foreign constraints.

Reply:
No, it doesn't. removed it

Reply:
Well, yes!
We need to add this and the foreign constraints are provided by the bundle, so no need to worry about that too!